packages/cloud-functions/src/cico/logParser/parseEventLogs.ts

```diff
@@ -0,0 +1,61 @@
+import { readFileSync } from 'fs'
+import { Providers } from '../Providers'
+
+// Inputs:
+const logFilePath = '/Users/tarik/Downloads/moonpay_logs.json' // Your local log file here
+const provider = Providers.Moonpay // The provider which is getting logs parsed
+
+const logs = JSON.parse(readFileSync(logFilePath, 'utf8'))
+
+const project = logs[0].logName.split('/')[1]
+process.env.GCLOUD_PROJECT = project
+// Using an in-line require because process variables need to be defined
+// before calling these functions
+const { parseMoonpayEvent, MOONPAY_BIG_QUERY_EVENT_TABLE } = require('../moonpayWebhook')
+const { parseRampEvent, RAMP_BIG_QUERY_EVENT_TABLE } = require('../rampWebhook')
+const { parseXanpoolEvent, XANPOOL_BIG_QUERY_EVENT_TABLE } = require('../xanpoolwebhook')
+const { parseTransakEvent, TRANSAK_BIG_QUERY_EVENT_TABLE } = require('../transakWebhook')
+
+const selectEventParserAndTable = (
+  provider: Providers
+): undefined | { parser: Function; table: string } => {
+  switch (provider) {
```

Comment:
Something I like to do in cases like this is create an object and evaluate it directly like this instead of the switch statement:
```
return {
  [Providers.Moonpay]: { parser: parseMoonpayEvent, table: MOONPAY_BIG_QUERY_EVENT_TABLE },
  [Providers.Ramp]: { parser: parseRampEvent, table: RAMP_BIG_QUERY_EVENT_TABLE },
  [Providers.Xanpool]: { parser: parseXanpoolEvent, table: XANPOOL_BIG_QUERY_EVENT_TABLE },
  [Provicers.Transak]: { parser: parseTransakEvent, table: TRANSAK_BIG_QUERY_EVENT_TABLE },
}[provider]
```

totally optional of course, I just like the way it ends up looking :)

Reply:
Oh yeah that's more compact, thanks for the suggestion. It's a bit more annoying in this case because not every provider has a property (there is no key for Simplex because it's not a webhook and doesn't really have logs to parse) so typescript complains.